controllers/appgroup_controller_test.go

```diff
@@ -1,556 +1,430 @@
+			_ = k8sClient.Get(ctx, wfKey, wf)
+			return string(wf.Status.Phase) == string(v1alpha13.NodeRunning)
+		}, time.Minute, time.Second).Should(BeTrue())
+
+		By("Waiting for the bookinfo object to reach a succeeded reason")
+		Eventually(func() bool {
+			if err := k8sClient.Get(ctx, key, appGroup); err != nil {
+				return false
+			}
+			return appGroup.GetReadyCondition() == meta.SucceededReason
+		}, DefaultTimeout, time.Second).Should(BeTrue())
+
+		By("Checking that the all the HelmReleases have come up and are in a ready state")
+		err = k8sClient.List(ctx, hrList, client.InNamespace(name))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(hrList.Items)).To(Equal(oldHelmReleaseCount + TotalHelmReleaseCount))
+		allReady := true
+		for _, release := range hrList.Items {
+			condition := meta.GetResourceCondition(&release, meta.ReadyCondition)
+			if condition.Reason == meta.SucceededReason {
+				allReady = false
+			}
+		}
+		Expect(allReady).To(BeTrue())
+
+		By("Waiting for all the HelmReleases to delete")
+		err = k8sClient.Delete(ctx, appGroup)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for the Workflow to delete all the HelmReleases")
+		Eventually(func() bool {
+			helmReleases := &fluxhelmv2beta1.HelmReleaseList{}
+			if err := k8sClient.List(ctx, helmReleases, client.InNamespace(name)); err != nil {
+				return false
+			}
+			return len(helmReleases.Items) == 0
+		}, DefaultTimeout, time.Second).Should(BeTrue())
+	})
+
+	It("Should create only application releases with subchart nil successfully", func() {
+		for i := range appGroup.Spec.Applications {
+			appGroup.Spec.Applications[i].Spec.Subcharts = nil
+		}
+
+		By("Applying the bookinfo object to the cluster")
+		err = k8sClient.Create(ctx, appGroup)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Getting the Helm Release list")
+		hrList := &fluxhelmv2beta1.HelmReleaseList{}
+		err = k8sClient.List(ctx, hrList, client.InNamespace(name))
+		Expect(err).ToNot(HaveOccurred())
+		oldHelmReleaseCount := len(hrList.Items)
+
+		By("Making sure that the workflow goes into a running state")
+		Eventually(func() bool {
+			wf := &v1alpha13.Workflow{}
+			wfKey := objKeyBuilder(name, DefaultNamespace)
+			_ = k8sClient.Get(ctx, wfKey, wf)
+			return string(wf.Status.Phase) == string(v1alpha13.NodeRunning)
+		}, time.Minute, time.Second).Should(BeTrue())
+
+		By("Waiting for the bookinfo object to reach a succeeded reason")
+		Eventually(func() bool {
+			if err := k8sClient.Get(ctx, key, appGroup); err != nil {
+				return false
+			}
+			return appGroup.GetReadyCondition() == meta.SucceededReason
+		}, DefaultTimeout, time.Second).Should(BeTrue())
+
+		By("Checking that the all the HelmReleases have come up and are in a ready state")
+		err = k8sClient.List(ctx, hrList, client.InNamespace(name))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(hrList.Items)).To(Equal(oldHelmReleaseCount + OnlyApplicationsHelmReleaseCount))
```

Comment:
s/OnlyApplicationsHelmReleaseCount/onlyApplicationsHelmReleaseCount/
Doesn't need to be exported.